R/write.R

```diff
@@ -149,56 +215,298 @@ add_labelled_xlsx_features <- function(supertbl,
+                x = supertbl_meta_labels, colNames = FALSE)
   }
 
   # Define redcap_data variable labels
   var_labels <- supertbl$redcap_data %>% map(\(x) generate_dictionaries(x))
 
   for (i in seq_along(supertbl$redcap_form_name)) {
-    openxlsx::writeData(wb = wb,
-                        sheet = sheet_vals[i],
-                        x = var_labels[[i]], colNames = FALSE)
+    wb$add_data(
+      sheet = sheet_vals[i],
+      x = var_labels[[i]], colNames = FALSE)
+  }
+
+  for (i in seq_len(nrow(wb$tables))) {
+    dims <- gsub("[0-9]+", "1", wb$tables$tab_ref[i])
 
-    # add style to labels
-    openxlsx::addStyle(wb,
-                       sheet = sheet_vals[i],
-                       rows = 1,
-                       cols = seq_along(var_labels[[i]]),
-                       style = label_header_style)
+    wb$add_cell_style(sheet = i,
+                      dims = dims,
+                      wrapText = "1")
+    wb$add_font(sheet = i,
+                dims = dims,
+                color = openxlsx2::wb_color(hex = "7F7F7F"),
+                italic = "1")
   }
 }
+
+#' @title Add the supertbl table of contents sheet
+#'
+#' @description
+#' Internal helper function. Adds appropriate elements to wb object. Returns
+#' a dataframe.
+#'
+#' @param supertbl a supertibble generated using `read_redcap()`
+#' @param wb An `openxlsx2` workbook object
+#' @param include_metadata Include a sheet capturing the combined output of the
+#' supertibble `redcap_metadata`. Default `TRUE`.
+#' @param labelled Whether or not to include labelled outputs, default `FALSE`.
+#' Requires use of `make_labelled()`.
+#' @param table_style Any excel table style name or "none" (see "formatting"
+#' vignette in \link[openxlsx2]{wb_add_data_table}). Default "tableStyleLight8".
+#' @param set_col_widths Width to set columns across the workbook. Default
+#' "auto", otherwise a numeric value. Standard Excel is 8.43.
+#' @param na_strings Value used for replacing NA values from x. Default
+#' `na_strings()` uses the special ⁠#N/A⁠ value within the workbook.
+#'
+#' @importFrom dplyr select mutate row_number across
+#' @importFrom tidyselect any_of
+#'
+#' @returns A dataframe
+#'
+#' @keywords internal
+
+add_supertbl_toc <- function(wb,
+                             supertbl,
+                             include_metadata,
+                             labelled,
+                             table_style,
+                             set_col_widths,
+                             na_strings) {
+
+  # To avoid XLSX indicators of "Number stored as text", change class type
+  convert_percent <- function(x) {
+    class(x) <- c("numeric", "percentage")
+    x
+  }
+
+  supertbl_toc <- supertbl %>%
+    # Remove list elements
+    select(-any_of(c("redcap_data", "redcap_metadata", "redcap_events"))) %>%
+    # Necessary to avoid "Number stored as text" Excel dialogue warnings
+    mutate(across(any_of("data_na_pct"), convert_percent))
+
```

Comment:
Can we convert the column that shows the size in memory to character so it appropriately says "kB", "MB", etc.?

Reply:
This should be resolved now in the [recent commit](https://github.com/CHOP-CGTInformatics/REDCapTidieR/pull/148/commits/06ac05b53d3d2cd8b14512594a8127ec727f4052).